README.md

```diff
@@ -276,4 +277,49 @@ Note that the error type is `Array String`, as we want to collect multiple error
 <!-- NOTE: we may want to link to https://book.purescript.org/chapter7.html#applicative-validation-1. -->
 
 Exercises:
+
 1. Migrate all the parsers to the new `Parser` type. That should be trivial, as it's only a matter of wrapping `Either`s with `V` or using `invalid` and `pure` instead of `Left` and `Right` respectively.
+
+#### Pill 7
+
+Now our rudimentary command-line options parser is mostly complete functionally. But some of you may find the boilerplate to define the `Parser` type's instances a bit annoying. In this isolated case, it's not much, but in a real-world application, where you would have different compositions of types (e.g. `Array` and `Maybe`), each will probably have instances similar to the others'. Wouldn't it be great if we had a type that would give us those instances for free?
+
+PureScript comes again to the rescue, this time with the `Compose` data type. It allows you to compose types together, giving you several instances as an extra (no `Bind`/`Monad` though; more on that later!). It's defined like this:
+
+```purescript
+newtype Compose f g a = Compose (f (g a))
+```
+
+Let's play around with it and add some instances!
+
+Exercises:
+
```

Comment:
Do you think that suggesting a concrete `f` and `g` as a helper could be sensible? I mean:

"(hint: If you have trouble finding implementation of these instances maybe try to think about composition of some concrete data types which are `Functor` / `Applicatives` and are familiar to you like `Maybe`, `Array` or `Function` etc.)"